spec/etl/item/content/parser_spec.rb

```diff
@@ -465,21 +465,38 @@ def build_raw_json(body:, schema_name:)
       end
     end
 
-    context 'when invalid schema' do
-      before do
-        expect(GovukError).to receive(:notify).with(an_instance_of(InvalidSchemaError))
-      end
+    context "when invalid schema" do
+      describe "has no schema_name and no base_path" do
+        before do
+          expect(GovukError).to receive(:notify).with(InvalidSchemaError.new("Schema does not exist: "), extra: { base_path: "" })
```

Comment:
These tests are fine as is, but I would structure these two a bit differently.

The reason being that tests are easier for me to understand if they follow a very simple structure

- setup
- do a thing
- assertions

The second two are really important for understanding what a test is doing, so I'd rather have them in the body of the test, even if it means duplicating some code.

With `expect(foo).to receive(:bar)` the order is reversed but I still think of it as an assertion rather than setup, so I'd rather inline this like in the tests below. Otherwise it's easy to miss the `before` and think the test isn't asserting anything.

The other thing that seems strange to me is the test names, because "it raised InvalidSchemaError" and "it returns nil" are mutually exclusive if you're talking about the same thing. The problem is there are two 'it's!

There is an error that gets raised internally but nothing outside the class depends on that behaviour - the thing you care about is really the side effect of calling `GovukError` rather than the exception itself, so the name of the spec should reflect that.

Maybe something like this?

```ruby
context "when the content has no schema_name and no base path" do
  describe '#extract_content' do
    it "returns nil and notifies us about the invalid schema" do
       ...
    end
  end
end
```